.github/workflows/ci.yml

```diff
@@ -206,16 +207,59 @@ jobs:
       with:
         action: none
         xcode: ^16
+
     - name: build and copy bindings
       run: |
         ./node-uniffi/build-ios.sh
         ./examples/ios/LuminaDemo/copy_bindings.sh
+
+    - name: create and boot simulator
+      run: |
+        xcrun simctl shutdown all || true
+        xcrun simctl erase all || true
+        xcrun simctl delete CI-iPhone || true
+
+        # Run `xcrun simctl list devicetypes` for more
+        DEVTYPE=com.apple.CoreSimulator.SimDeviceType.iPhone-16
+        # Run `xcrun simctl list runtimes` for more
+        RUNTIME=com.apple.CoreSimulator.SimRuntime.iOS-26-0
+
+        # Create simulator
+        UDID=$(xcrun simctl create CI-iPhone "$DEVTYPE" "$RUNTIME")
+        # Boot it
+        xcrun simctl boot "$UDID"
+        # Wait until fully booted
+        xcrun simctl bootstatus "$UDID" -b
+
+        # Set env for the next steps
+        echo "SIM_UDID=$UDID" >> "$GITHUB_ENV"
+
     - name: xcode clean build
       working-directory: examples/ios/LuminaDemo
-      run: xcodebuild -scheme LuminaDemo clean build CODE_SIGN_IDENTITY="" CODE_SIGNING_REQUIRED=NO
+      env:
+        NSUnbufferedIO: "YES"
```

Comment:
what is this for?

Reply:
no hard feelings, was just wondering what's that to see if we want to comment on it why we want it, but I'm good after your explanation